frontend/src/metabase/public/LocaleProvider.unit.spec.tsx

```diff
@@ -97,3 +103,67 @@ describe("getLocaleToUse", () => {
     });
   });
 });
+
+describe("LocaleProvider", () => {
+  it("should make Mantine components use correct locale", async () => {
+    mockLocaleJsonResponse("de");
+
+    renderWithProviders(
+      <LocaleProvider locale="de">
+        <DatePicker defaultDate={new Date(2020, 0, 1)} onChange={() => {}} />
+      </LocaleProvider>,
+    );
+
+    await waitForLocaleJson("de");
+
+    // `waitFor` to ensure the component has time to re-render
+    await waitFor(() => {
+      expect(screen.getByText("Januar 2020")).toBeInTheDocument();
+    });
+  });
+
+  it("should make useLocale return the correct locale", async () => {
+    const TestComponent = () => {
+      const locale = useLocale();
+      return <div>{locale}</div>;
+    };
+
+    mockLocaleJsonResponse("de");
+
+    renderWithProviders(
+      <LocaleProvider locale="de">
+        <TestComponent />
+      </LocaleProvider>,
+    );
+
+    await waitForLocaleJson("de");
+
+    // `waitFor` to ensure the component has time to re-render
+    await waitFor(() => {
+      expect(screen.getByText("de")).toBeInTheDocument();
+    });
+  });
+});
```

Comment:
I suggest adding a test that verifies how we handle hyphenated locales like `pt-BR`. I noticed some strange behavior around this earlier this week. Sometimes our locales use underscores instead of hyphens.

Reply:
Terrific!